pkg/apis/operator/v1alpha1/helper/garden.go

```diff
@@ -168,3 +173,40 @@ func HighAvailabilityEnabled(garden *operatorv1alpha1.Garden) bool {
 func TopologyAwareRoutingEnabled(settings *operatorv1alpha1.Settings) bool {
 	return settings != nil && settings.TopologyAwareRouting != nil && settings.TopologyAwareRouting.Enabled
 }
+
+// GetMinAllowedForKubeAPIServer returns the minAllowed configuration values for Kube API Server if configured or nil otherwise.
+func GetMinAllowedForKubeAPIServer(apiServerConfig *operatorv1alpha1.KubeAPIServerConfig) corev1.ResourceList {
+	if apiServerConfig != nil {
+		return v1beta1helper.GetMinAllowedForKubeAPIServer(apiServerConfig.KubeAPIServerConfig)
+	}
+	return nil
+}
+
+// GetMinAllowedForETCDMain returns the minAllowed configuration values for etcd main if configured or nil otherwise.
+func GetMinAllowedForETCDMain(etcd *operatorv1alpha1.ETCD) corev1.ResourceList {
+	return copyMinAllowed(func() corev1.ResourceList {
+		if etcd != nil && etcd.Main != nil && etcd.Main.Autoscaling != nil {
+			return etcd.Main.Autoscaling.MinAllowed
+		}
+		return nil
+	})
+}
+
+// GetMinAllowedForETCDEvents returns the minAllowed configuration values for etcd events if configured or nil otherwise.
+func GetMinAllowedForETCDEvents(etcd *operatorv1alpha1.ETCD) corev1.ResourceList {
+	return copyMinAllowed(func() corev1.ResourceList {
+		if etcd != nil && etcd.Events != nil && etcd.Events.Autoscaling != nil {
+			return etcd.Events.Autoscaling.MinAllowed
+		}
+		return nil
+	})
+}
+
+func copyMinAllowed(getMinAllowed func() corev1.ResourceList) corev1.ResourceList {
+	if minAllowed := getMinAllowed(); minAllowed != nil {
+		minAllowedCopy := make(corev1.ResourceList)
+		maps.Copy(minAllowedCopy, minAllowed)
```

Comment:
Isn't possible to use `maps.Clone` in the same way how it is used in the v1beta1 helper?